src/hotspot/share/nmt/mallocTracker.cpp

```diff
@@ -239,13 +243,17 @@ bool MallocTracker::print_pointer_information(const void* p, outputStream* st) {
   const MallocHeader* likely_live_block = nullptr;
   {
     const size_t smallest_possible_alignment = sizeof(void*);
-    const uint8_t* here = align_down(addr, smallest_possible_alignment);
-    const uint8_t* const end = here - (0x1000 + sizeof(MallocHeader)); // stop searching after 4k
+    uintptr_t here = (uintptr_t)align_down(addr, smallest_possible_alignment);
+    if (here == 0) {
+      return false; // bail out
+    }
+    uintptr_t end = (here > (0x1000 + sizeof(MallocHeader)))
+                      ? here - (0x1000 + sizeof(MallocHeader)) // stop searching after 4k
+                      : 0;
```

Comment:
`here` will always be > 0, overflowed or not. You need to check the decrement amount before decrementing. With "capping end" I meant setting it at or above the decrement amount, e.g.
```
end = MAX2(smallest_possible_alignment, here - (0x1000 + sizeof(MallocHeader))
```
(sorry if I wasn't clear)

Reply:
oh, I saw 0 and thought that I need to consider that as minimum value possible. Sorry. I have updated code now.